assets/js/components/SettingsNotice/SettingsNoticeWithIntersectionObserver.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Settings notice component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useIntersection } from 'react-use';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import SettingsNotice from './SettingsNotice';
+
+// TODO: This could renamed/moved to be more generic e.g. `ComponentWithIntersectionObserver`,
+// or refactored to a HOC.
+function SettingsNoticeWithIntersectionObserver( {
+	onInView,
+	...settingsNoticeProps
+} ) {
+	const trackingRef = useRef();
```

Comment:
What do you think about changing this variable name to something else, as this isn't specific to **tracking** only. How about `componentRef` or `inViewRef`?

Reply:
Good suggestion @nfmohit , have renamed this, bearing in mind this component will be moving and changing names as part of work I am currently in progress with.

Reply:
Perfect, thanks @10upsimon !